pex/common.py

```diff
@@ -334,4 +337,8 @@ def delete(self):
   def zip(self, filename, mode='wb'):
     with contextlib.closing(zipfile.ZipFile(filename, mode)) as zf:
       for f in sorted(self.files()):
-        zf.write(os.path.join(self.chroot, f), arcname=f, compress_type=zipfile.ZIP_DEFLATED)
+        filepath = os.path.join(self.chroot, f)
+        perm = oct(os.stat(filepath)[stat.ST_MODE])[-3:]
```

Comment:
FWICT there is no need to `oct` the value here into a string only to `int` it below. Just `perm = os.stat(filepath).st_mode`, then `fileinfo.external_attr = perm << 16`.